data/Kubiak_2009/raw/Kubiak_2009_extra_R_code.R

```diff
@@ -0,0 +1,246 @@
+      str_extract(primary_juvenile_period, "[0-9\\-]+(?=w)")    
+    ),
+    fire_time_from_fire_to_flowering = primary_juvenile_period,
+    secondary_juvenile_period = case_when(
+      str_detect(secondary_juvenile_period, "\\+") ~ str_extract(secondary_juvenile_period, "[0-9\\-]+(?=\\+)"),
+      str_detect(secondary_juvenile_period, "y") ~ str_extract(secondary_juvenile_period, "[0-9\\-][0-9\\-\\.]*(?=y)"),
+      TRUE ~ str_extract(secondary_juvenile_period, "[0-9\\-]+(?=w)")
+    ),
+    peak_flowering_resp = if_else(
+      str_detect(peak_flowering_resp, "\\+"),
+      str_extract(peak_flowering_resp, "[0-9\\-]+(?=\\+)"),
+      str_extract(peak_flowering_resp, "[0-9][0-9\\-\\.]*(?=y)")    
+    ),
+    peak_flowering_seed = if_else(
+      str_detect(peak_flowering_seed, "\\+"),
+      str_extract(peak_flowering_seed, "[0-9\\-]+(?=\\+)"),
+      str_extract(peak_flowering_seed, "[0-9][0-9\\-\\.]*(?=y)")    
+    ),
+    peak_flowering_unknown = if_else(
+      str_detect(peak_flowering_unknown, "\\+"),
+      str_extract(peak_flowering_unknown, "[0-9\\-]+(?=\\+)"),
+      str_extract(peak_flowering_unknown, "[0-9][0-9\\-\\.]*(?=y)")    
+    ),
+    peak_fruiting_resp = if_else(
+      str_detect(peak_fruiting_resp, "\\+"),
+      str_extract(peak_fruiting_resp, "[0-9\\-]+(?=\\+)"),
+      str_extract(peak_fruiting_resp, "[0-9][0-9\\-\\.]*(?=y)")    
+    ),
+    peak_fruiting_seed = if_else(
+      str_detect(peak_fruiting_seed, "\\+"),
+      str_extract(peak_fruiting_seed, "[0-9\\-]+(?=\\+)"),
+      str_extract(peak_fruiting_seed, "[0-9][0-9\\-\\.]*(?=y)")    
+    ),
+    peak_fruiting_unknown = if_else(
+      str_detect(peak_fruiting_unknown, "\\+"),
+      str_extract(peak_fruiting_unknown, "[0-9\\-]+(?=\\+)"),
+      str_extract(peak_fruiting_unknown, "[0-9][0-9\\-\\.]*(?=y)")    
+    ),
+    large_prop_flowering_resp = if_else(
+      str_detect(large_prop_flowering_resp, "y"),
+      str_extract(large_prop_flowering_resp, "[0-9][0-9\\-\\.]*(?=y)"),
+      str_extract(large_prop_flowering_resp, "[0-9\\-]+(?=w)")
+    ),
+    large_prop_flowering_seed = if_else(
+      str_detect(large_prop_flowering_seed, "y"),
+      str_extract(large_prop_flowering_seed, "[0-9][0-9\\-\\.]*(?=y)"),
+      str_extract(large_prop_flowering_seed, "[0-9\\-]+(?=w)")
+    ),
+    large_prop_flowering_unknown = if_else(
+      str_detect(large_prop_flowering_unknown, "y"),
+      str_extract(large_prop_flowering_unknown, "[0-9][0-9\\-\\.]*(?=y)"),
+      str_extract(large_prop_flowering_unknown, "[0-9\\-]+(?=w)")
+    ),
+    advanced_seedlings = if_else(is.na(advanced_seedlings) & !is.na(seedlings_first_observed), 
+                                 "just germinated seedlings", advanced_seedlings)
+  ) -> wide_data
+
+# Find ranges to split into min and max columns
+#wide_data$first_fruiting_seed %>% unique -> unique
+#unique[unique %>% str_detect("-")]
+
+# Move ranges to min and max columns and remove from original columns
+wide_data %>% 
```

Comment:
I'm glad you're so fast at stringr coding! This is a ton of work

Reply:
My partner helped me with some of it 😅 and the more I do it the faster I get!